onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.cc

```diff
@@ -526,13 +542,19 @@ Status GreedySearchProcessLogits(
                                          cudaMemcpyHostToDevice, cuda_stream));
   }
 
+  // TODO(hasesh): Can we avoid the const_cast by changing the interface of
+  // GreedySearchProcessLogits() to take in a non-const OrtValue for logits
+  // as this is the only place we will ever use the logits and it may be reasonable
+  // to allow this method to mutate/process the logits in-place
   cuda::LaunchLogitsProcessKernel<CudaT>(
```

Comment:
> cuda::LaunchLogitsProcessKernel<CudaT>( [](http://example.com/codeflow?start=1&length=40)

We can set the logits value in range [vocab_size, padded_vocab_size] to smallest, and then you don't need to change the TopK

Reply:
In LaunchLogitsProcessKernel() ?

Reply:
Yes